influxdb3_wal/src/object_store.rs

```diff
@@ -458,29 +476,79 @@ impl FlushBuffer {
     /// responses. If a snapshot should occur with this flush, a semaphore permit is also returned.
     async fn flush_buffer_into_contents_and_responses(
         &mut self,
-    ) -> (
+        force_snapshot: bool,
+    ) -> Option<(
         WalContents,
         Vec<oneshot::Sender<WriteResult>>,
-        Option<(SnapshotInfo, OwnedSemaphorePermit)>,
-    ) {
+        Option<(SnapshotDetails, OwnedSemaphorePermit)>,
+    )> {
+        // whether wal buffer is empty or not, we can force snapshot. But when it's empty
+        // we need to add some WalOp to capture the snapshot details in wal file and
+        // currently WalOp::Noop is used
+        let forced_snapshot_details = if self.wal_buffer.is_empty() && force_snapshot {
+            debug!(">>> adding a no op with force snapshot details");
+            self.wal_buffer.add_no_op();
+            self.snapshot_tracker.snapshot(force_snapshot)
+        } else if force_snapshot {
```

Comment:
I think you want to lose this else if and change the else down below to just call `self.snapshot_tracker.snapshot(force_snapshot)`